manager/manager.go

```diff
@@ -616,20 +613,21 @@ func (m *Manager) handleLeadershipEvents(ctx context.Context, leadershipCh chan
 }
 
 // serveListener serves a listener for local and non local connections.
-func (m *Manager) serveListener(ctx context.Context, errServe chan error, proto string, lis net.Listener) {
+func (m *Manager) serveListener(ctx context.Context, errServe chan error, lis listener) {
 	ctx = log.WithLogger(ctx, log.G(ctx).WithFields(
 		logrus.Fields{
-			"proto": lis.Addr().Network(),
-			"addr":  lis.Addr().String()}))
-	if proto == "unix" {
+			"proto": lis.l.Addr().Network(),
+			"addr":  lis.l.Addr().String(),
+		}))
+	if lis.local {
```

Comment:
I think Network or type assertion is a lesser evil. The new type is hard to follow, at least for me. For isLocal I think next code would work:
```
_, ok := l.(*net.TCPListener)
return !ok
```